### ISSUE TITLE
augmentation: RandomResizedCrop(scale=(1, 1)) never keeps a square or portrait image: strict candidate test and an inverted-ratio fallback

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4814]
`RandomResizedCrop` keeps the whole image under `scale=(1.0, 1.0)` only for landscape inputs. For a square or portrait input it returns a smaller crop that falls outside both `scale` and `ratio`. torchvision's reference `RandomResizedCrop.get_params` keeps the whole image in these cases.

Two lines in `kornia/augmentation/random_generator/_2d/crop.py` (on `main` 57cd68e6f) combine to cause this:

1. **The candidate test is strict** (line 268): `(w < size[1]) * (h < size[0])`. torchvision accepts `0 < w <= width and 0 < h <= height`. With `scale=(1.0, 1.0)` no candidate can be strictly smaller on both axes, so every call falls back.
2. **The fallback inverts the ratio** (lines 283-290). kornia's `ratio` is width/height (`w = sqrt(area * ratio)`), but the fallback computes `in_ratio = H / W`. It compares that against `min(ratio)` in both branches and never against `max(ratio)`. torchvision compares `W / H` against `min(ratio)` and `max(ratio)` and keeps the whole image when the input ratio is already in range.

## Repro

```python
import torch
import kornia.augmentation as K

def crop_hw(shape):
    torch.manual_seed(0)
    src = K.RandomResizedCrop((4, 4), scale=(1.0, 1.0), p=1.0).forward_parameters((1, 1, *shape))["src"][0]
    return int(src[2, 1] - src[1, 1]) + 1, int(src[1, 0] - src[0, 0]) + 1

for shape in [(8, 6), (8, 8), (32, 32), (100, 60), (60, 100)]:
    print(shape, crop_hw(shape))
```

Compared with torchvision 0.29.0's `RandomResizedCrop.get_params` run on the same shapes (the function taken verbatim from the wheel):

| input H x W | kornia crop | area | w/h | torchvision crop |
|---|---|---|---|---|
| 8 x 6 | 4 x 6 | 0.50 | 1.50 | 8 x 6 |
| 8 x 8 | 6 x 8 | 0.75 | 1.33 | 8 x 8 |
| 32 x 32 | 24 x 32 | 0.75 | 1.33 | 32 x 32 |
| 100 x 60 | 45 x 60 | 0.45 | 1.33 | 80 x 60 |
| 60 x 100 | 60 x 80 | 0.80 | 1.33 | 60 x 80 |

In the 8 x 6 row the crop's w/h of 1.5 is outside the default `ratio=(3/4, 4/3)`. It also has half the area that `scale` requests. Landscape inputs agree with torchvision only because the default `ratio` is symmetric (`min = 1 / max`).

## Expected

Follow torchvision's rule: a candidate may equal the input size, and the fallback compares `W / H` with both `min(ratio)` and `max(ratio)`, keeping the whole image when it is in range. Each change alone flips the current pin.

## Where it is documented

The `RandomResizedCrop` Convention block and `test_convention_random_resized_crop_fallback_can_escape_scale_and_ratio` currently describe this as intended behaviour. #4791 reframes both as a wart linked to this issue. The fix PR inverts the pin and drops the wart sentence.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by ducha-aiki** [comment]
## How often it hits

The fallback runs only when all 10 random candidates miss. The default ranges almost never trigger it; ranges close to "keep the whole image" do. The table shows the share of 20,000 sampled crops whose area falls below `scale[0]` (with 0.02 slack), on `main` at 750c1611e:

| `scale` | 224×224 (square) | 256×192 (portrait) | 192×256 (landscape) |
|---|---|---|---|
| `(0.08, 1.0)` default | 0.0% | 0.0% | 0.0% |
| `(0.5, 1.0)` | 0.0% | 0.0% | 0.0% |
| `(0.9, 1.0)` | 16.5% | 42.8% | 0.0% |
| `(1.0, 1.0)` | 100% | 100% | 0.0% |

Default ImageNet-style training is therefore unaffected. A "mostly resize, little crop" setting silently crops square and portrait inputs well below the requested area: `scale=(1.0, 1.0)` gives a 75% crop on a square image. Landscape inputs happen to be right because the default `ratio` is symmetric.

```python
import torch
import kornia.augmentation as K

torch.manual_seed(0)
n = 20000
for scale in [(0.08, 1.0), (0.5, 1.0), (0.9, 1.0), (1.0, 1.0)]:
    for H, W in [(224, 224), (256, 192), (192, 256)]:
        src = K.RandomResizedCrop((32, 32), scale=scale).forward_parameters((n, 3, H, W))["src"]
        h = src[:, 2, 1] - src[:, 1, 1] + 1
        w = src[:, 1, 0] - src[:, 0, 0] + 1
        area = h * w / (H * W)
        print(f"scale={scale} input {H}x{W}: {(area < scale[0] - 0.02).float().mean():.1%} of crops below scale[0]")
```

## For whoever picks this up

This is open for a contributor. The fix is small and self-contained:

1. In `RandomResizedCropGenerator.forward` (`kornia/augmentation/random_generator/_2d/crop.py`), accept candidates with `w <= W` and `h <= H`. Then make the fallback follow torchvision's `get_params`: compare `W / H` with both `min(ratio)` and `max(ratio)`, and keep the whole image when it is in range. Keep the branchless `torch.where` selection, because the generator is compiled with `fullgraph`.
2. Once #4791 is merged, remove the wart sentence that links #4814 from `RandomResizedCrop`'s Convention block. Replace `test_wart_random_resized_crop_fallback_leaves_scale_and_ratio_4814` in `tests/augmentation/test_conventions_geometric_crop.py` with a test of the correct behaviour: `scale=(1.0, 1.0)` keeps the whole image for square, portrait and landscape inputs. Run `grep -rnE "#4814|issues/4814" kornia/ tests/ docs/` to find every mention.
3. Add a `changelog.d/<PR>.fixed.md` fragment. The sampled crops change for configurations that hit the fallback, so say so in the fragment.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by adimalkar** [comment]
Opened #4823, following your plan: candidates may now equal the input (`<=`), the fallback follows `get_params` (`W / H` against both `min` and `max(ratio)`), the selection stays branchless and still compiles with `fullgraph=True`, and the wart pin is replaced by convention tests.

Two things the PR describes that you may want to see:

- **The default `scale`/`ratio` moves slightly too**: 9, 20 and 17 of 2000 draws on 224x224, 256x192 and 192x256. It isn't the fallback — every one of those draws is a crop that now spans a full input axis, i.e. the `<=` change, which torchvision also has. The `fixed` fragment says so.
- **Three seeded tests and the docstring example** use `ratio=(2, 2)` on a 3x3 input, which always falls back. They had pinned a 3-tall, 2-wide crop — the inverse of width / height = 2 — so they now expect the 2x3 crop torchvision gives. I checked the new values against an independent `F.interpolate` of that crop rather than copying them from the code.

The fallback crop is still placed at a random position, where torchvision centres it; that's outside this issue, so I only documented it.

(I used an AI assistant for this; details are in the PR's "How did AI help?" section.)
